editor/components/EditCrashAddressModal.tsx

```diff
@@ -0,0 +1,249 @@
+  rpt_street_desc: string | null;
+  rpt_road_part_id: number | null;
+  rpt_rdwy_sys_id: number | null;
+  rpt_hwy_num: string | null;
+  rpt_sec_block_num: string | null;
+  rpt_sec_street_pfx: string | null;
+  rpt_sec_street_name: string | null;
+  rpt_sec_street_sfx: string | null;
+  rpt_sec_street_desc: string | null;
+  rpt_sec_road_part_id: number | null;
+  rpt_sec_rdwy_sys_id: number | null;
+  rpt_sec_hwy_num: string | null;
+};
+
+/**
+ * Modal form component used for editing the crash address fields
+ */
+export default function EditCrashAddressModal({
+  setShowEditAddressModal,
+  crashId,
+  onSaveCallback,
+  show,
+  crash,
+}: EditCrashAddressModalProps) {
+  const defaultValues = useMemo(() => {
+    return (
+      crash && {
+        rpt_block_num: crash.rpt_block_num,
+        rpt_street_pfx: crash.rpt_street_pfx,
+        rpt_street_name: crash.rpt_street_name,
+        rpt_street_sfx: crash.rpt_street_sfx,
+        rpt_street_desc: crash.rpt_street_desc,
+        rpt_road_part_id: crash.rpt_road_part_id,
+        rpt_rdwy_sys_id: crash.rpt_rdwy_sys_id,
+        rpt_hwy_num: crash.rpt_hwy_num,
+        rpt_sec_block_num: crash.rpt_sec_block_num,
+        rpt_sec_street_pfx: crash.rpt_sec_street_pfx,
+        rpt_sec_street_name: crash.rpt_sec_street_name,
+        rpt_sec_street_sfx: crash.rpt_sec_street_sfx,
+        rpt_sec_street_desc: crash.rpt_sec_street_desc,
+        rpt_sec_road_part_id: crash.rpt_sec_road_part_id,
+        rpt_sec_rdwy_sys_id: crash.rpt_sec_rdwy_sys_id,
+        rpt_sec_hwy_num: crash.rpt_sec_hwy_num,
+      }
+    );
+  }, [crash]);
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    getValues,
+    setValue,
+    formState: { isDirty },
+  } = useForm<AddressFormInputs>({
+    defaultValues: defaultValues,
+  });
+
+  const { mutate } = useMutation(UPDATE_CRASH);
+
+  const primaryColumns = [
+    crashesColumns.rpt_block_num,
+    crashesColumns.rpt_street_pfx,
+    crashesColumns.rpt_street_name,
+    crashesColumns.rpt_street_sfx,
+    crashesColumns.rpt_street_desc,
+    crashesColumns.rpt_road_part_id,
+    crashesColumns.rpt_rdwy_sys_id,
+    crashesColumns.rpt_hwy_num,
+  ];
+  const secondaryColumns = [
+    crashesColumns.rpt_sec_block_num,
+    crashesColumns.rpt_sec_street_pfx,
+    crashesColumns.rpt_sec_street_name,
+    crashesColumns.rpt_sec_street_sfx,
+    crashesColumns.rpt_sec_street_desc,
+    crashesColumns.rpt_sec_road_part_id,
+    crashesColumns.rpt_sec_rdwy_sys_id,
+    crashesColumns.rpt_sec_hwy_num,
+  ];
+
+  /**
+   * Submits mutation to database on save button click
+   */
+  const onSave: SubmitHandler<AddressFormInputs> = async (data) => {
+    reset(data);
+    await mutate({
+      id: crashId,
+      updates: data,
+    });
+
+    onSaveCallback();
+  };
+
+  /**
+   * Component that renders the form field based on the input type
+   * of the column, such as text or select
+   */
+  const FormField = ({ col }: FormColumnProps) => {
```

Comment:
i didnt really want to pull this out into its own file but what do yall think?

Reply:
i went ahead and moved it into its own component called `AddressFormField` bc it fixed the error you described in another comment about the field losing focus. its true we have repeated this code a few times now :grimacing: i think kicking the can and addressing it app-wide is a good idea

also i see your point about the monolithic components folder, its getting hugeeee. could be another good improvement for the codebase in the future, to reorganize our folder structure